uxarray/grid/grid.py

```diff
@@ -2485,3 +2501,87 @@ def get_faces_between_latitudes(self, lats: Tuple[float, float]):
 
         """
         return faces_within_lat_bounds(lats, self.face_bounds_lat.values)
+
+    def get_faces_containing_point(self, point, tolerance=ERROR_TOLERANCE):
+        """Identifies the indices of faces that contain a given point.
+        Parameters
+        ----------
+        point : numpy.ndarray
+            A point in either cartesian coordinates or spherical coordinates
+        tolerance : numpy.ndarray
+            An optional error tolerance for points that lie on the nodes of a face
+
+        Returns
+        -------
+        index : numpy.ndarray
+            Array of the face indices containing point. Empty if no face is found
+
+        """
```

Comment:
Maybe mention something along the lines of:
- This function will typically return a single face, unless the point falls directly on a corner or edge, where there will instead be multiple values.